src/main/java/nwoolcan/controller/warehouse/WarehouseControllerImpl.java

```diff
@@ -83,19 +90,50 @@ public IngredientArticleViewModel createIngredientArticle(final String name,
     }
 
     @Override
-    public Result<StockViewModel> createStock(final int articleId, final Date expirationDate) {
+    public Result<PlainStockViewModel> createStock(final int articleId, final Date expirationDate) {
         return Result.of(articleId)
                      .flatMap(this::getArticleById)
                      .flatMap(article -> warehouse.createStock(article, expirationDate))
-                     .map(StockViewModel::new);
+                     .map(PlainStockViewModel::new);
     }
 
     @Override
-    public Result<StockViewModel> createStock(final int articleId) {
+    public Result<PlainStockViewModel> createStock(final int articleId) {
         return Result.of(articleId)
                      .flatMap(this::getArticleById)
                      .flatMap(warehouse::createStock)
-                     .map(StockViewModel::new);
+                     .map(PlainStockViewModel::new);
+    }
+
+    @Override
+    public Result<AbstractArticleViewModel> setName(final int articleId, final String newName) {
+        return Result.of(articleId)
+                     .flatMap(this::getArticleById)
+                     .flatMap(article -> warehouse.setName(article, newName))
+                     .map(AbstractArticleViewModel::getViewArticle);
+    }
+
+    @Override
+    public Result<Empty> addRecord(final int stockId, final double amount, final Record.Action action, final Date date) {
+        final Result<Stock> stockResult = getStockById(stockId);
+        return stockResult.flatMap(stock -> Quantity.of(amount, stock.getArticle().getUnitOfMeasure()))
+                          .flatMap(quantity -> stockResult.getValue().addRecord(new Record(quantity, date, action)))
+                          .toEmpty();
+    }
+
+    @Override
+    public Result<Empty> addRecord(final int stockId, final double amount, final Record.Action action) {
+        return addRecord(stockId, amount, action, new Date());
+    }
+
+    @Override
+    public Result<AbstractArticleViewModel> getViewArticleById(final int articleId) {
+        return getArticleById(articleId).map(AbstractArticleViewModel::getViewArticle);
+    }
+
+    @Override
+    public Result<DetailStockViewModel> getViewStockById(final int stockId) {
+        return getStockById(stockId).map(DetailStockViewModel::getDetailViewStock);
```

Comment:
use new `ControllerUtils` class for these :)

Reply:
useless, i'm the only who use them